index.js

```diff
@@ -70,8 +61,14 @@ module.exports = (input, opts) => {
 				task: () => del('node_modules')
 			},
 			{
-				title: `Installing dependencies${opts.yarn ? ' using Yarn' : ''}`,
-				task: install(bin, opts.yarn)
+				title: 'Installing dependencies using Yarn',
+				enabled: () => opts.yarn === true,
+				task: () => exec('yarn', ['install'])
```

Comment:
> Difficult to detect the binary is missing versus a standard error with the current library 

You can easily check for `ENOENT`

> If the binary is missing in that scenario, we shouldn't fallback to npm because it could silently break the build due to using non-pinned versions

Did you know that a lockfile does nothing in a dependency? What I mean is that if I install (nom or yarn) `execa` which has a `yarn.lock` file, the lockfile is not taken into account for the dependencies of `execa`. So actually the only thing a lock file does is ensuring the same dependencies between contributors, not between contributors and its users. So falling back to `npm` is not a bad thing at all because you would end up with the exact same dependencies as your users! This is an important thing to keep in mind.

So yes, I still think we should fall back to npm if yarn does not exist. You can easily set  a custom message in Listr to notify the users that yarn is not installed and that it's falling back to nom because of that. We can even recommend the users that they should install yarn, but we should not be the one that should enforce yarn upon our users. If they don't want it they don't want it, their choice.